sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -71,6 +103,7 @@ public virtual void StopTestRecording()
             save |= SaveDebugRecordingsOnFailure;
 #endif
             Recording?.Dispose(save);
+            Logger?.OutputEventsForTest();
```

Comment:
Why don't we directly log to TestContext.Log?

Reply:
Do you mean do away with the Logger abstraction?

Reply:
No, keep the abstraction just lose the StringBuilder

Reply:
We'll need the `StringBuilder` to attached failed test output to the test if we're not logging to stdout. While having to look over successful tests' output is hopefully rare, looking at failed tests' output in the UI is going to be much more common. Please keep this the way it is. We can tee to a file for all failed and successful tests and let it write to the TRX file.

Reply:
I'm not sure I follow. We will always be writing to TestContext.Out, regardless of whether test passed or failed. For passed tests, this will not be attached to the individual tests within dev ops, but you can download the directory of trx files to see the output.

Reply:
Seems we were referring to different StringBuilders.